src/api/callApi.ts

```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+export const getBook = async () => {
+  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
+    query:
+      'query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}',
+  });
+  return response.data;
+};
+
+export const getChapter = async (bookId: number) => {
```

Comment:
Could you type the return value?

Reply:
```
interface BookApiResponse {
  data: {
    viewer: {
      books: {
        hits: bookType[];
      };
    };
  };
}
export const getBook = async (): Promise<BookApiResponse> => {
  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
    query:
      'query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}',
  });
  return response.data;
};
```

Reply:
Cool. Now, can you create a generic type that can handle both `books` and `chapters`?

Reply:
I create a new branch with fix.
I have left store but => AxiosError: Request failed with status code 503

Reply:
If you get API errors, it's useful to capture a `curl` that can be used as a reproduction of the error. Many factors can lead to an API error, and without knowing the request that was sent, it's hard to infer what went wrong.

The two sample requests do return the expected data.